### PR TITLE
#1097: [Typo] TransitionWrapper (closes #1097)

### LINKED ISSUE #1097: [Typo] TransitionWrapper
[Project card](https://github.com/buildo/react-components/projects/1?fullscreen=true#card-4470804)

← #1034

## requirements
{describe the new feature}

## specs
{optional: describe technical specs to implement this feature, if not obvious}

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by veej**
Closes #1097

## Test Plan

> These results can be screenshots, query results, or even just pointers to unit tests that successfully passed.

- [ ] Google Chrome
- [ ] Internet Explorer

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.

### CODE REVIEW

File: src/TransitionWrapper/TransitionWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    leave?: React.CSSProperties,
+    leaveActive?: React.CSSProperties
+  }
+}
+
+export type TransitionWrapperDefaultProps<CP> = {
+  /** object with inline-style for each transition event. It's also possible to use `css` classes (formatted in kebab-case) */
+  transitionStyles: TransitionWrapperProps.TransitionStyles,
+  /** custom component to be used as wrapper for `children`. Can be either an html tag name string (eg. 'div', 'span', etc), or a `ReactClass` (eg. `FlexView`) */
+  component:  keyof React.ReactHTML | React.ComponentClass<CP>,
```
Comment (veej) [inline src/TransitionWrapper/TransitionWrapper.tsx]: Qui abbiamo un buco: se component è un `keyof React.ReactHTML`, così non viene enforzato che le sue props siano `CP`

File: src/TransitionWrapper/TransitionWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  /** the component you want to animate (it must have a unique "key") */
+  children: JSX.Element,
+  /** duration of enter transition in milliseconds */
+  transitionEnterTimeout: number,
+  /** duration of leave transition in milliseconds */
+  transitionLeaveTimeout: number,
+  className?: string,
+};
+
+export type TransitionWrapperProps<CP> = TransitionWrapperRequiredProps & Partial<TransitionWrapperDefaultProps<CP>> & {[k in keyof CP]: CP[k]};
```
Comment (veej) [inline src/TransitionWrapper/TransitionWrapper.tsx]: sostituendo {[k in keyof CP]: CP[k]} con un più ovvio CP, abbiamo errori del typechecker, che non riesce a inferire i tipi delle props

**Comment by giogonzo** [comment]
screen shot 2017-09-06 at 10 57 43 am

here I expect `href` to be ok as a prop.. but it isn't
Not sure if I'm missing something or if the component typing is :thinking:

**Comment by veej** [comment]
as discussed offline, using `<React.AnchorHTMLAttributes<HTMLAnchorElement>>` in the second example will make it work. Not sure if we can do better

**Pull request merged**